idaes/generic_models/unit_models/distillation/tray_column.py

```diff
@@ -0,0 +1,311 @@
+                                      self.config.has_vapor_side_draw,
+                                      "has_heat_transfer":
+                                      self.config.has_heat_transfer,
+                                      "has_pressure_change":
+                                      self.config.has_pressure_change,
+                                      "property_package":
+                                      self.config.property_package,
+                                      "property_package_args":
+                                      self.config.property_package_args}})
+
+        # Add condenser
+        self.condenser = Condenser(
+            default={"condenser_type": self.config.condenser_type,
+                     "temperature_spec":
+                         self.config.condenser_temperature_spec,
+                     "property_package": self.config.property_package,
+                     "property_package_args":
+                     self.config.property_package_args})
+
+        # Add reboiler
+        self.reboiler = Reboiler(
+            default={"has_boilup_ratio": True,
+                     "has_pressure_change": self.config.has_pressure_change,
+                     "property_package": self.config.property_package,
+                     "property_package_args":
+                     self.config.property_package_args})
+
+        # Construct arcs between trays, condenser, and reboiler
+        self._make_arcs()
+        TransformationFactory("network.expand_arcs").apply_to(self)
+
+    def _make_arcs(self):
+        # make arcs
+        self.liq_stream_index = RangeSet(0, self.config.number_of_trays)
+        self.vap_stream_index = RangeSet(1, self.config.number_of_trays + 1)
+
+        def rule_liq_stream(self, i):
+            if i == 0:
+                return {"source": self.condenser.reflux,
+                        "destination": self.tray[i + 1].liq_in}
+            elif i == self.config.number_of_trays:
+                return {"source": self.tray[i].liq_out,
+                        "destination": self.reboiler.inlet}
+            else:
+                return {"source": self.tray[i].liq_out,
+                        "destination": self.tray[i + 1].liq_in}
+
+        def rule_vap_stream(self, i):
+            if i == 1:
+                return {"source": self.tray[i].vap_out,
+                        "destination": self.condenser.inlet}
+            elif i == self.config.number_of_trays + 1:
+                return {"source": self.reboiler.vapor_reboil,
+                        "destination": self.tray[i - 1].vap_in}
+            else:
+                return {"source": self.tray[i].vap_out,
+                        "destination": self.tray[i - 1].vap_in}
+
+        self.liq_stream = Arc(self.liq_stream_index, rule=rule_liq_stream)
+        self.vap_stream = Arc(self.vap_stream_index, rule=rule_vap_stream)
+
+    def propagate_stream_state(self, source=None,
```

Comment:
I think we have a utility function that already does this  in `idaes.core.util.initialization`.

Reply:
I had to write my own because the utility function did not support indexed arcs.

Reply:
My idea is to spin off these and the `make_phase_split` methods as distillation only utility methods.

Reply:
As a thought, maybe we should update the utility method to optionally handle indexed Arcs then?

Reply:
Considered it but thought will merge later..